### PR TITLE
refactor(uniswap-v2): Changed protobuf types from byte arrays to strings for ease of debugging

### PR DESCRIPTION
**Pull request by this-username-is-taken**

### CODE REVIEW

File: common/proto/common.proto [hunk 1]
```diff
@@ -7,14 +7,14 @@ message ERC20Tokens {
 }
 
 message ERC20Token {
-  bytes address = 1; // bytes: Address
+  string address = 1; // bytes: Address
```
Comment (0xbe1) [inline common/proto/common.proto:10]: update comment

File: common/proto/common.proto [hunk 2]
```diff
@@ -7,14 +7,14 @@ message ERC20Tokens {
...
   string name = 2;
   string symbol = 3;
   uint64 decimals = 4;
 }
 
 message Transaction {
-  bytes hash = 1; // bytes: Hash
+  string hash = 1; // bytes: Hash
```
Comment (0xbe1) [inline common/proto/common.proto:17]: update comment

File: common/proto/dex_amm.proto [hunk 3]
```diff
@@ -6,15 +6,15 @@ package messari.dex_amm.v1;
 
 message Pool {
   string name = 1;
-  bytes address = 2; // bytes: Address
+  string address = 2; // bytes: Address
```
Comment (0xbe1) [inline common/proto/dex_amm.proto:9]: update comment

File: common/proto/dex_amm.proto [hunk 4]
```diff
@@ -6,15 +6,15 @@ package messari.dex_amm.v1;
...
   repeated messari.common.v1.ERC20Token input_tokens = 3;
 
   // Metrics
   string total_value_locked = 100; // string: BigDecimal
 }
 
 message PoolToken {
-  bytes address = 1; // bytes: Address
+  string address = 1; // bytes: Address
```
Comment (0xbe1) [inline common/proto/dex_amm.proto:17]: update comment

File: erc20-price/src/lib.rs [hunk 5]
```diff
@@ -32,7 +32,7 @@ fn map_price_for_tokens(
         prices.items.push(Erc20Price {
             block_number,
             price_usd: token_price.to_string(),
-            token_address: erc20_token.clone(),
+            token_address: Hex(erc20_token.clone()).to_string(),
```
Comment (0xbe1) [inline erc20-price/src/lib.rs:35]: `hex::encode(erc20_token.clone())`

File: uniswap-v2/proto/v2/uniswap.proto [hunk 6]
```diff
@@ -7,12 +7,12 @@ package messari.uniswap.v2;
...
-  bytes tx_hash = 1; // bytes: Hash
+  string tx_hash = 1; // bytes: Hash
   uint32 log_index = 2;
 
-  bytes token0 = 3; // bytes: Address
-  bytes token1 = 4; // bytes: Address
-  bytes pair = 5; // bytes: Address
+  string token0 = 3; // bytes: Address
+  string token1 = 4; // bytes: Address
+  string pair = 5; // bytes: Address
```
Comment (0xbe1) [inline uniswap-v2/proto/v2/uniswap.proto:15]: update comments

File: uniswap-v2/proto/v2/uniswap.proto [hunk 7]
```diff
@@ -21,10 +21,10 @@ message PairCreatedEvents {
 
 // event Mint(address indexed sender, uint amount0, uint amount1);
 message MintEvent {
-  bytes tx_hash = 1; // bytes: Hash
+  string tx_hash = 1; // bytes: Hash
   uint32 log_index = 2;
 
-  bytes sender = 3; // bytes: Address
+  string sender = 3; // bytes: Address
```
Comment (0xbe1) [inline uniswap-v2/proto/v2/uniswap.proto:27]: update comments

File: uniswap-v2/proto/v2/uniswap.proto [hunk 8]
```diff
@@ -35,13 +35,13 @@ message MintEvents {
...
   string amount0 = 4; // string: BigInt
   string amount1 = 5; // string: BigInt
-  bytes to = 6; // bytes: Address
+  string to = 6; // bytes: Address
```
Comment (0xbe1) [inline uniswap-v2/proto/v2/uniswap.proto:44]: update comments

File: uniswap-v2/proto/v2/uniswap.proto [hunk 9]
```diff
@@ -50,15 +50,15 @@ message BurnEvents {
...
 
-  bytes sender = 3; // bytes: Address
+  string sender = 3; // bytes: Address
   string amount0In = 4; // string: BigInt
   string amount1In = 5; // string: BigInt
   string amount0Out = 6; // string: BigInt
   string amount1Out = 7; // string: BigInt
-  bytes to = 8; // bytes: Address
+  string to = 8; // bytes: Address
 }
```
Comment (0xbe1) [inline uniswap-v2/proto/v2/uniswap.proto:62]: update comments

File: uniswap-v2/src/lib.rs [hunk 10]
```diff
@@ -47,10 +47,10 @@ fn map_pair_created_event(
...
 
             pair_created_events.items.push(uniswap::PairCreatedEvent {
-                token0: event.token0,
-                token1: event.token1,
-                pair: event.pair,
-                tx_hash: log.receipt.transaction.clone().hash,
+                token0: Hex(event.token0).to_string(),
+                token1: Hex(event.token1).to_string(),
+                pair: Hex(event.pair).to_string(),
+                tx_hash: Hex(log.receipt.transaction.clone().hash).to_string(),
```
Comment (0xbe1) [inline uniswap-v2/src/lib.rs:53]: `hex::encode(...)`

File: common/proto/common.proto (hunk 1 above)
Comment (this-username-is-taken) [reply common/proto/common.proto:10]: I actually left them in on purpose so we can easily change back after SF adds better support. Even a comment like `bytes: Address` makes it more clear that the string is going to be addresses formatted in hex.

File: erc20-price/src/lib.rs (hunk 5 above)
Comment (this-username-is-taken) [reply erc20-price/src/lib.rs:35]: what's the difference here? Seems like both do the same thing

Comment (0xbe1) [reply erc20-price/src/lib.rs:35]: not different function-wise, just shorter and less overhead

Hex(...) creates a Hex struct first, then call to_string() method
while Hex::encode(...) is like calling a static method, thus no struct created, thus less overhead

**Review by 0xbe1 — APPROVED** [review]
other than Hex(...).to_string() => Hex::encode(...), lgtm

**Pull request merged**